components/network/latency/metrics.go

```diff
@@ -0,0 +1,100 @@
+package latency
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/leptonai/gpud/components"
+	components_metrics_state "github.com/leptonai/gpud/pkg/gpud-metrics/state"
+	"github.com/leptonai/gpud/pkg/log"
+	components_metrics "github.com/leptonai/gpud/pkg/metrics"
+)
+
+const SubSystem = "network_latency"
+
+var (
+	lastUpdateUnixSeconds = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "",
+			Subsystem: SubSystem,
+			Name:      "last_update_unix_seconds",
+			Help:      "tracks the last update time in unix seconds",
+		},
+	)
+
+	edgeInMilliseconds = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "",
+			Subsystem: SubSystem,
+			Name:      "edge_in_milliseconds",
+			Help:      "tracks the edge latency in milliseconds",
+		},
+		[]string{"provider_region"},
+	)
+)
+
+var _ components.PromRegisterer = (*component)(nil)
+
+func (c *component) RegisterCollectors(reg *prometheus.Registry, dbRW *sql.DB, dbRO *sql.DB, tableName string) error {
+	c.metricsMu.Lock()
+	defer c.metricsMu.Unlock()
+
+	c.edgeInMillisecondsMetricsStore = components_metrics.NewStore(dbRW, dbRO, tableName, SubSystem+"_edge_in_milliseconds")
```

Comment:
It's not reasonable for creating metric store for each metric..  We should invert the dependency.. We can implement a global metric store, then implement a scraper or collector to collect all these metrics from prometheus.Registry, so that we can save all these metrics to the store.. And we can easily change the underlay store to TSDB in future...